src/autoIncrement.ts

```diff
@@ -94,7 +93,7 @@ export function AutoIncrementID(schema: mongoose.Schema<any>, options: AutoIncre
       return;
     }
 
-    const modelName = opt.model || (this.constructor as any).modelName;
+    const modelName = (this.constructor as any).modelName;
```

Comment:
```suggestion
    const modelName: string = (this.constructor as any).modelName;
```
just to not have an any type